flit_core/flit_core/common.py

```diff
@@ -348,7 +348,7 @@ class Metadata(object):
     requires_external = ()
     provides_extra = ()
```

Comment:
```suggestion
    provides_extra = ()
    dynamic = ()
```

Let's add the field for completeness even if we're not using it. Also in the `write_metadata_file()` method, so we don't overlook it if there ever is a need for it.